libcloud/loadbalancer/drivers/dimensiondata.py

```diff
@@ -0,0 +1,741 @@
+
+from libcloud.common.dimensiondata import DimensionDataConnection
+from libcloud.common.dimensiondata import DimensionDataPool
+from libcloud.common.dimensiondata import DimensionDataPoolMember
+from libcloud.common.dimensiondata import DimensionDataVirtualListener
+from libcloud.common.dimensiondata import DimensionDataVIPNode
+from libcloud.common.dimensiondata import API_ENDPOINTS
+from libcloud.common.dimensiondata import DEFAULT_REGION
+from libcloud.common.dimensiondata import TYPES_URN
+from libcloud.utils.misc import reverse_dict
+from libcloud.utils.xml import fixxpath, findtext, findall
+from libcloud.loadbalancer.types import State
+from libcloud.loadbalancer.base import Algorithm, Driver, LoadBalancer
+from libcloud.loadbalancer.base import Member
+from libcloud.loadbalancer.types import Provider
+
+
+class DimensionDataLBDriver(Driver):
+    """
+    DimensionData node driver.
+    """
+
+    selected_region = None
+    connectionCls = DimensionDataConnection
+    name = 'Dimension Data Load Balancer'
+    website = 'https://cloud.dimensiondata.com/'
+    type = Provider.DIMENSIONDATA
+    api_version = 1.0
+
+    network_domain_id = None
+
+    _VALUE_TO_ALGORITHM_MAP = {
+        'ROUND_ROBIN': Algorithm.ROUND_ROBIN,
+        'LEAST_CONNECTIONS': Algorithm.LEAST_CONNECTIONS,
+        'SHORTEST_RESPONSE': Algorithm.SHORTEST_RESPONSE,
+        'PERSISTENT_IP': Algorithm.PERSISTENT_IP
+    }
+    _ALGORITHM_TO_VALUE_MAP = reverse_dict(_VALUE_TO_ALGORITHM_MAP)
+
+    _VALUE_TO_STATE_MAP = {
+        'NORMAL': State.RUNNING,
+        'PENDING_ADD': State.PENDING,
+        'PENDING_CHANGE': State.PENDING,
+        'PENDING_DELETE': State.PENDING,
+        'FAILED_ADD': State.ERROR,
+        'FAILED_CHANGE': State.ERROR,
+        'FAILED_DELETE': State.ERROR,
+        'REQUIRES_SUPPORT': State.ERROR
+    }
+
+    def __init__(self, key, secret=None, secure=True, host=None, port=None,
+                 api_version=None, region=DEFAULT_REGION, **kwargs):
+
+        if region not in API_ENDPOINTS:
+            raise ValueError('Invalid region: %s' % (region))
+
+        self.selected_region = API_ENDPOINTS[region]
+
+        super(DimensionDataLBDriver, self).__init__(key=key, secret=secret,
+                                                    secure=secure, host=host,
+                                                    port=port,
+                                                    api_version=api_version,
+                                                    region=region,
+                                                    **kwargs)
+
+    def _ex_connection_class_kwargs(self):
+        """
+            Add the region to the kwargs before the connection is instantiated
+        """
+
+        kwargs = super(DimensionDataLBDriver,
+                       self)._ex_connection_class_kwargs()
+        kwargs['region'] = self.selected_region
+        return kwargs
+
+    def create_balancer(self, name, port, protocol, algorithm, members):
```

Comment:
Is `members` a required argument or you can create a load-balancer with no initial members and attach them later?